src/uct/sm/scopy/base/scopy_ep.c

```diff
@@ -154,22 +145,34 @@ ucs_arbiter_cb_result_t uct_scopy_ep_progress_tx(ucs_arbiter_t *arbiter,
                                                 arbiter);
     uct_scopy_ep_t *ep       = ucs_container_of(ucs_arbiter_elem_group(elem),
                                                 uct_scopy_ep_t, arb_group);
-    uct_scopy_tx_t *tx       = ucs_container_of(elem, uct_scopy_tx_t, arb_elem);
-    size_t seg_size          = ucs_min(iface->config.seg_size,
-                                       tx->total_length - tx->consumed_length);
     unsigned *count          = (unsigned*)arg;
+    uct_scopy_tx_arb_elem_t *tx_arb_elem;
+    uct_scopy_tx_t *tx;
+    uct_scopy_tx_comp_t *flush_comp;
     ucs_status_t status;
-
-    status = iface->tx(&ep->super.super, tx->iov, tx->iov_cnt,
-                       &tx->iov_iter, &seg_size,
-                       tx->remote_addr + tx->consumed_length,
-                       tx->rkey, tx->op);
-    status = uct_scopy_ep_comp_tx(ep, tx, status, seg_size);
-    if (ucs_likely(!UCS_STATUS_IS_ERR(status))) {
-        count++;
-        if (status == UCS_INPROGRESS) {
-            return UCS_ARBITER_CB_RESULT_RESCHED_GROUP;
+    size_t seg_size;
+
+    tx_arb_elem = ucs_container_of(elem, uct_scopy_tx_arb_elem_t, arb_elem);;
+    if (tx_arb_elem->op != UCT_SCOPY_TX_FLUSH_COMP) {
+        ucs_assert((tx_arb_elem->op == UCT_SCOPY_TX_GET_ZCOPY) ||
+                   (tx_arb_elem->op == UCT_SCOPY_TX_PUT_ZCOPY));
+        tx       = ucs_container_of(elem, uct_scopy_tx_t, super.arb_elem);
```

Comment:
maybe instead o defining uct_scopy_tx_comp_t just use uct_scopy_tx_t for flush ops as well? it already has "comp", and they come from same mpool anyway so we don't really save memory

Reply:
done